### PR TITLE
Add API documentation (#172)

### PR DESCRIPTION
**Pull request by ErikMichelson**
Adding documentation for the available REST endpoints in markdown and OpenAPIv3 format.
Some endpoints are not included. These are mainly login (and callbacks)/logout and quite internal things.
See #172

### CODE REVIEW

File: docs/dev/openapi.yml [hunk 1]
```diff
@@ -188,7 +188,294 @@ paths:
...
+    post:
+      tags:
+        - user
+      summary: Replace user's history with a new one.
+      description: The body must be form-encoded and contain a field `history` with a JSON-encoded array like its returned from the server when exporting the history.
+      requestBody:
+        required: true
+        content:
+          'application/x-www-form-urlencoded':
+            example: 'history=[{"id":"example","text":"Untitled","time":1556275442010,"tags":[],"pinned":false]'
```
Comment (SISheogorath) [inline docs/dev/openapi.yml]: The JSON here is invalid

File: docs/dev/openapi.yml [hunk 2]
```diff
@@ -0,0 +1,481 @@
...
+          in: path
+          required: true
+          description: The note for which the pinned state should be toggled
+          content:
+            'text/plain':
+              example: my-note
+    delete:
+      tags:
+        - user
+      summary: Deletes a note from the user's history.
```
Comment (SISheogorath) [inline docs/dev/openapi.yml]: Either this or the rest of this section is wrong, since this talks about note deletion, the rest about pinned notes.

Comment (ErikMichelson) [reply docs/dev/openapi.yml]: I just tested again the requests made by the frontend. A POST request to the endpoint needs the pinned=true/false body, a DELETE request removes the note from the history. Omitting the pinned=true/false body while doing a POST request to the endpoint does not create the history entry but returns a HTTP 500. I'll investigate the code behind this.

Comment (ErikMichelson) [reply docs/dev/openapi.yml]: The methods being called by the history-router are `historyDelete` and `historyPost` in the lib/history.js.

The [`historyPost`](https://github.com/codimd/server/blob/ce99a79352ea145cfbfc579592b96e53ba843b66/lib/history.js#L154-L170) method differentiates between the endpoints with note id and without. In the part for the endpoint with the note id only the toggle for the pinned state will be executed.

The [`historyDelete`](https://github.com/codimd/server/blob/master/lib/history.js#L184-L194) method differentiates similar and in the relevant part for this endpoint it just deletes the note from the user's history.

Comment (SISheogorath) [reply docs/dev/openapi.yml]: We should really work on a proper `/api/` endpoint collection :see_no_evil: 

Thanks a lot for your investigative work here!

Comment (ErikMichelson) [reply docs/dev/openapi.yml]: Yes we should. But documenting the current status is at least something.

**Pull request merged**